gophers/yoda_gopher.go

```diff
@@ -0,0 +1,55 @@
+package gophers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// YodaGopher replies with Yoda transformation of text quotes instead of relaying the original message.
+type YodaGopher struct {
+	endpoint string
+}
+
+func NewYodaGopher() YodaGopher {
+	return YodaGopher{
+		endpoint: "https://yoda.p.mashape.com/yoda",
+	}
+}
+
+func ReadMashapeKey() string {
+	key, err := ioutil.ReadFile("MashapeKey")
+	if err != nil {
+		fmt.Println("No MashapeKey file present.")
+	}
+	return strings.Replace(string(key), "\n", "", -1)
+}
+
+func (g YodaGopher) TransformMessage(msg string) string {
+	client := &http.Client{Timeout: 40 * time.Second}
+	request, _ := http.NewRequest("GET", g.endpoint, nil)
+
+	request.Header.Add("X-Mashape-Key", ReadMashapeKey())
+	request.Header.Add("Accept", "text/plain")
+
+	query := request.URL.Query()
```

Comment:
Since the URL doesn't have anything in the querystring, it probably makes more sense to just init a blank one instead?

```golang
query := url.Values{}
```